### PR TITLE
server: proxy: improve modules api

### PR DESCRIPTION
**Pull request by kubistika**
This one improves the modularity of the old filters api (which renamed to modules API).

It allows external modules to register server channels in runtime (we need this for multi factor authentication system we're building), closing the connection and etc.

Also cleans the code a bit and removes the filters (modules) list from `proxyConfig*` struct, which wasn't the location for this list from the beginning.

### CODE REVIEW

File: server/proxy/modules/modules_api.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	proxyHookFn ServerChannelsInit;
+	proxyHookFn ServerChannelsFree;
+
+	/* proxy filters a module can set these function pointers to register filters. */
+	proxyFilterFn KeyboardEvent;
+	proxyFilterFn MouseEvent;
+};
+
+/* filter events parameters */
+#pragma pack(push, 1)
```
Comment (akallabeth) [inline server/proxy/modules/modules_api.h]: Check these, there are some defines for packing alignment (see `pack.h`)

**Comment by kubistika** [comment]
@akallabeth your "nitpicking" is always welcome and helpful! I'll fix soon

File: server/proxy/modules/modules_api.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	proxyHookFn ServerChannelsInit;
+	proxyHookFn ServerChannelsFree;
+
+	/* proxy filters a module can set these function pointers to register filters. */
+	proxyFilterFn KeyboardEvent;
+	proxyFilterFn MouseEvent;
+};
+
+/* filter events parameters */
+#define WINPR_PACK_PUSH
```
Comment (akallabeth) [inline server/proxy/modules/modules_api.h]: Ok, sorry, looks like you found a header that has a few bugs :/

File: server/proxy/pf_context.c [hunk 3]
```diff
@@ -141,14 +148,6 @@ rdpContext* p_client_context_create(rdpSettings* clientSettings)
...
 
-static void connection_info_free(connectionInfo* info)
-{
-	free(info->TargetHostname);
-	free(info->ClientHostname);
-	free(info->Username);
-	free(info);
-}
-
 proxyData* proxy_data_new()
```
Comment (akallabeth) [inline server/proxy/pf_context.c]: missing `(void)`

File: server/proxy/pf_context.h [hunk 4]
```diff
@@ -93,17 +96,13 @@ struct proxy_data
...
-
-	connectionInfo* info;
-	filters_list* filters;
 };
 
 /* client */
 rdpContext* p_client_context_create(rdpSettings* clientSettings);
 
 /* pdata */
 proxyData* proxy_data_new();
```
Comment (akallabeth) [inline server/proxy/pf_context.h]: missing `(void)`

**Comment by kubistika** [comment]
@akallabeth anything else?

**Pull request merged**